src/tileset.rs

```diff
@@ -0,0 +1,189 @@
+    /// Tile width and height.
+    pub fn tile_size(&self) -> &TileSize {
+        &self.tile_size
+    }
+    /// Number to show in the UI for the tile with index=0. Default is 1.
+    /// Only used for Aseprite UI purposes. Not used for data representation.
+    pub fn base_index(&self) -> &i16 {
+        &self.base_index
+    }
+    pub fn name(&self) -> &String {
+        &self.name
+    }
+    /// When Some, includes a link to an external file.
+    pub fn external_file(&self) -> &Option<ExternalTilesetReference> {
+        &self.external_file
+    }
+    /// TODO: Tileset image export
+    pub fn image(&self) -> RgbaImage {
+        todo!()
+    }
+
+    pub(crate) fn parse_chunk(data: &[u8], pixel_format: PixelFormat) -> Result<Tileset> {
+        let mut reader = AseReader::new(data);
+        let id = reader.dword().map(|val| TilesetId(val))?;
+        let flags = reader.dword().map(|val| TilesetFlags { bits: val })?;
+        let empty_tile_is_id_zero = flags.contains(TilesetFlags::EMPTY_TILE_IS_ID_ZERO);
+        let tile_count = reader.dword()?;
+        let tile_width = reader.word()?;
+        let tile_height = reader.word()?;
+        let tile_size = TileSize {
+            width: tile_width,
+            height: tile_height,
+        };
+        let base_index = reader.short()?;
+        // Reserved bytes
+        reader.skip_bytes(14)?;
+        let name = reader.string()?;
+        let external_file = {
+            if !flags.contains(TilesetFlags::LINKS_EXTERNAL_FILE) {
+                None
+            } else {
+                Some(ExternalTilesetReference::parse(&mut reader)?)
+            }
+        };
+        let pixels = {
+            if !flags.contains(TilesetFlags::FILE_INCLUDES_TILES) {
+                None
+            } else {
+                let _compressed_length = reader.dword()?;
+                let expected_pixel_count =
+                    (tile_count * (tile_height as u32) * (tile_width as u32)) as usize;
+                Pixels::from_compressed(reader, pixel_format, expected_pixel_count).map(Some)?
+            }
+        };
+        Ok(Tileset {
+            id,
+            empty_tile_is_id_zero,
+            tile_count,
+            tile_size,
+            base_index,
+            name,
+            external_file,
+            pixels,
+        })
+    }
+}
+#[derive(Debug)]
+pub struct TilesetsById(HashMap<TilesetId, Tileset>);
+impl TilesetsById {
+    pub(crate) fn new() -> Self {
+        Self(HashMap::new())
+    }
+    pub(crate) fn add(&mut self, tileset: Tileset) {
+        self.0.insert(*tileset.id(), tileset);
+    }
+    pub fn map(&self) -> &HashMap<TilesetId, Tileset> {
+        &self.0
+    }
+}
+impl Index<TilesetId> for TilesetsById {
+    type Output = Tileset;
+    fn index(&self, id: TilesetId) -> &Self::Output {
+        let map = self.map();
+        if map.contains_key(&id) {
+            return &self.map()[&id];
+        }
+        panic!("no external file found for id")
```

Comment:
Hm, again not a fan of this panic. I guess in this case, it's possible for the user to avoid it? I think it would be better to just provide a `get` method and not an `Index` impl. Then the user has to decide whether to call `unwrap` or not.